### ISSUE TITLE
0.0.5 causes truffle migrate to hang

### ISSUE THREAD

**Issue by timothywangdev** [trufflesuite/truffle-hdwallet-provider#34]
# Steps to reproduce:
```
ubuntu 16.04
truffle@4.1.8
nodejs@v9.11.1
npm@5.6.0
```

```
mkdir test
cd test
truffle init
npm install truffle-hdwallet-provider@0.0.5
```

Use the following truffle.js

```
const HDWalletProvider = require('truffle-hdwallet-provider')

let mnemonic = 'fantasy achieve abandon weekend ability fashion media guilt message island identify much'
var provider = new HDWalletProvider(mnemonic, "https://rinkeby.infura.io/[ID]");


module.exports = {
  // See <http://truffleframework.com/docs/advanced/configuration>
  // for more about customizing your Truffle configuration!
  networks: {
    development: {
      host: '127.0.0.1',
      port: 8545,
      network_id: '*', // Match any network id
      gas: 5000000
    }
  },
  mocha: {
    reporter: 'eth-gas-reporter',
    reporterOptions : {
      currency: 'USD'
    }
  }
}

```

# Run migration
```
truffle migrate
```

# Output
```
Using network 'development'.

Running migration: 1_initial_migration.js
  Replacing Migrations...
  ... 0xae7895ce2a393747fbcc57b6b05b2e5ec5ce50204e0b3fdc664aa4bfdef490de
  Migrations: 0xe8b2c5ca5f6fb180da5396cbd6c11724bf5cce81
Saving successful migration to network...
  ... 0x9eed1277d362ee410a6496fae9c6983da441da8fda19198b24389e736e82f302
Saving artifacts...
```

It hangs while saving artifacts, have to use ^C to terminate it.

Note that 0.0.4 works fine.

**Comment by chuckbergeron** [comment]
I had this same issue, where 0.0.5 was hanging at the end of `truffle compile` and `truffle migrate` (`truffle console` then `compile` and `migrate` in there worked fine).

I attempted to roll-back to 0.0.4, but then `truffle migrate` was failing with "invalid JSON RPC response".

Finally, tested going back to 0.0.3 and now everything works great.

**Comment by cgewecke** [comment]
Apologies for this - the last release contains a significant `provider-engine` and unfortunately it was causing truffle commands to hang right as they exit.  Bug is fixed on the truffle nightly, [darq-truffle](https://www.npmjs.com/package/darq-truffle) and will be in a patch release scheduled for early next week.

**Comment by cgewecke** [comment]
Should be fixed with truffle `4.1.9`, please ping if you continue to see this.
